src/Dataset.ts

```diff
@@ -474,29 +478,46 @@ export default class Dataset {
         stringifyQueryObj.stringify({ limit: 50, ...pick(payload, "subject", "predicate", "object", "graph") }),
     });
   }
-  public async uploadAsset(fileOrPath: string | File, assetName?: string): Promise<Asset> {
+  public async uploadAsset(fileOrPath: string | File, opts: { mode: UploadAssetModes; name?: string }): Promise<Asset>;
```

Comment:
@dionenak `opts` and `mode` should be optional as a user should just be able to supply a name.

Reply:
@ivan-kovatchev I don't understand why we should do this. I made opts and mode required because I want to promote people being more explicit about what they want to happen. If they don't want to use mode, they can still use the other interface, but it is not recommended.

Reply:
Our convention for TriplyDB-JS is to make it easy for users to interact with TriplyDB. This means not making everything explicit, as that's more work for users to type/remember/maintain. To that end, we should have sensible defaults for function arguments.

For example, the `getAsset()` function ([docs](https://docs.triply.cc/triplydb-js/dataset/#datasetgetassetname-string-version-number)) returns the latest version of the asset by default, but you can be more explicit and specify a version to return. If you had to do this all the time, it would make the function tedious to use.

For that same reason, we should have a default mode for how the `uploadAsset()` function works, which is intuitive and documented.

Reply:
I think that the example you showed is an oversimplification of the problem, and it is not that similar to uploadAsset().

I don't see how throwing is a sensible default in any case. And I couldn't decide for a sensible default between replacing and adding a new version. This is why I do think that the user setting the mode should be required.

Nevertheless, I believe that you have more experience with this library and it's use than I do, so I will follow your advice.